### ISSUE TITLE
TypeError: undefined is not a function in  at t.default.c.value ([...]/mobx-react-devtools/index.js:1:9788)

### ISSUE THREAD

**Issue by jieter** [mobxjs/mobx-react-todomvc#7]
hoi @mweststrate 

I'm trying to get this example working, not having any react experience, but I'm having a couple of issues:

- My [atom](http://atom.io) starts complaining about eslint plugins missing. This is because may global eslint is old, fixed by installing eslint locally: `npm install --save-dev eslint babel-eslint`. Note that the lint tests report some problems.
- After running `npm start`, waiting some time and browsing to `localhost:4000`, I just get `Server error`, with the following in my console:
```
jieter@zimri:~/workspace/mobx-react-todomvc (master)$ npm start

> mobx-react-todomvc@1.0.0 start /home/jieter/workspace/mobx-react-todomvc
> node bin/server.js

Listening on port 4000
webpack built 42dd9e7008f4e665a8d6 in 2174ms
Error on request GET /
TypeError: undefined is not a function
    at t.default.c.value (/home/jieter/workspace/mobx-react-todomvc/node_modules/mobx-react-devtools/index.js:1:9788)
    at t.render (/home/jieter/workspace/mobx-react-todomvc/node_modules/mobx-react-devtools/index.js:1:13340)
    at [object Object].ReactCompositeComponentMixin._renderValidatedComponentWithoutOwnerOrContext (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactCompositeComponent.js:587:34)
    at [object Object].ReactCompositeComponentMixin._renderValidatedComponent (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactCompositeComponent.js:607:32)
    at [object Object].wrapper [as _renderValidatedComponent] (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactPerf.js:66:21)
    at [object Object].ReactCompositeComponentMixin.mountComponent (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactCompositeComponent.js:220:30)
    at [object Object].wrapper [as mountComponent] (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactPerf.js:66:21)
    at Object.ReactReconciler.mountComponent (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactReconciler.js:37:35)
    at ReactDOMComponent.ReactMultiChild.Mixin.mountChildren (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactMultiChild.js:241:44)
    at ReactDOMComponent.Mixin._createContentMarkup (/home/jieter/workspace/mobx-react-todomvc/node_modules/react/lib/ReactDOMComponent.js:591:32)
```

Am I doing something wrong?

```
$ node -v
v0.12.4
```

**Comment by mweststrate** [comment]
Hej gast, goed jouw hier te zien :)

I've just tried to reproduce the issue with a fresh checkout but it seems to start fine. Would you mind checking whether a newer node version (4+) fixes the issue?

**Comment by jieter** [comment]
Also tried [mobx-react-typescript](https://github.com/contacts-mvc/mobx-react-typescript), which worked without problems on node 0.12.4. This works like charm with 6.0.0. 

Want to keep it open for the devDependency issues?

**Comment by jieter** [comment]
Also note that I can't seem to find any documented minimum requirement for the node version, other than you removing them from the [travis configs](https://github.com/mobxjs/mobx/commit/f29d3aebe9cf1f2099dd7fbeb22f53553c49db99).

**Comment by mweststrate** [comment]
There is no specific dependency on a node version, MobX and related packages run on any ES5 compliant javascript engine (see [faq](http://mobxjs.github.io/mobx/faq/faq.html)), which is nowadays practically everything except IE 8.

It was removed from the travis configs in the MobX repo for more flexibility in the test-suite. So the dependency is probably coming from one of the dev deps in this repo. I updated the readme to reflect that, as it is not particularly interesting to support node 3 major versions back in an example project ;-).

**Comment by mweststrate** [comment]
Btw you could try an older version of this repo: https://github.com/mobxjs/mobx-react-todomvc/commit/0061c12fd24ba5b4dd0d5d123482af57cb7fbf77 which doesn't include server side rendering and such. Or use `nvm` to run multiple node versions simultaneously on the same machine.

**Comment by jieter** [comment]
'3 major versions back' might be a long time in your perspective, but 0.10.25 is the default for Ubuntu 15.10 I'm running, which is not really old to me. Apparently, I upgraded to nodejs 0.12.4 after installing.

Anyway, a note like you just added can save people frustration which is what you want in an example project I guess?

What about the eslint/babel-eslint deps?

**Comment by mweststrate** [comment]
Just checked the transitive dependencies, the minimum listed required version is node 3, so `npm install` should have warned about that in your case. But in general these messages are well hidden between all the other output.

eslint deps is optional. 

In general, the standard provided packages by Ubuntu for node are really behind and quite unsuitable for any practical development. I suggest using the official node channels instead: https://nodejs.org/en/download/package-manager/#debian-and-ubuntu-based-linux-distributions

**Comment by andykog** [comment]
I've just added polyfill for Object.assign in https://github.com/mobxjs/mobx-react-devtools/commit/8122c42b01e9b8d08c19ca6a82dbc38f86fe3940
It will work with node v0.12.4 after next version of mobx-react-devtools is published

**Comment by mweststrate** [comment]
Thx @andykog for figuring that out! I've just released it as `4.0.5`.

@jieter could you verify whether this solves the issue? (I just updated package json of this repo already). Note that the `react` package itself depends on `fbjs` which requires `node >= 3`, so it is possible that the above issue is solved but that you encounter a new one.
